src/legacy/core_plugins/kibana_react/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -66,53 +63,47 @@ export function TopNavMenu(props: Props) {
   }
 
   function renderSearchBar() {
-    // Validate presense of all required fields
+    const { uiSettings, http, notifications, savedObjects } = kibana.services;
+
+    // If any fields are missing, render nothing
     if (
       !props.showSearchBar ||
-      !props.savedObjects ||
-      !props.http ||
-      !props.notifications ||
+      !savedObjects ||
+      !http ||
+      !notifications ||
+      !uiSettings ||
       !props.timeHistory
     )
       return;
```

Comment:
@lizozom This condition was silently causing my usage of `TopNav` to fail without noticeable errors. I hope that your continued work on `TopNav` will improve this!